src/partially_blindrsa.ts

```diff
@@ -0,0 +1,489 @@
+        //    and stop
+        const c = is_coprime(m, n);
+        if (c === false) {
+            throw new Error('invalid input');
+        }
+
+        // 7. r = random_integer_uniform(1, n)
+        const r = random_integer_uniform(n, kLen);
+
+        // 8. inv = inverse_mod(r, n)
+        // 9. If inverse_mod fails, raise a "blinding error" error
+        //    and stop
+        let inv: Uint8Array;
+        try {
+            inv = i2osp(r.inverseMod(n), kLen);
+        } catch (e) {
+            throw new Error(`blinding error: ${(e as Error).toString()}`);
+        }
+
+        // 10. pk_derived = DerivePublicKey(pk, info)
+        const pk_derived = await this.derivePublicKey(pk, info);
+
+        // 11. x = RSAVP1(pk, r)
+        const x = rsavp1(pk_derived, r);
+
+        // 12. z = m * x mod n
+        const z = m.mulmod(x, n);
+
+        // 13. blinded_msg = int_to_bytes(z, modulus_len)
+        const blindedMsg = i2osp(z, kLen);
+
+        // 14. output blinded_msg, inv
+        return { blindedMsg, inv };
+    }
+
+    async blindSign(
+        privateKey: CryptoKey,
+        blindMsg: Uint8Array,
+        info: Uint8Array,
+    ): Promise<Uint8Array> {
+        const { jwkKey, modulusLengthBytes: kLen } = await this.extractKeyParams(
+            privateKey,
+            'private',
+        );
+        if (!jwkKey.n || !jwkKey.d) {
+            throw new Error('key has invalid parameters');
+        }
+        const n = sjcl.bn.fromBits(sjcl.codec.base64url.toBits(jwkKey.n));
+        const d = sjcl.bn.fromBits(sjcl.codec.base64url.toBits(jwkKey.d));
+        const p = sjcl.bn.fromBits(sjcl.codec.base64url.toBits(jwkKey.p));
+        const q = sjcl.bn.fromBits(sjcl.codec.base64url.toBits(jwkKey.q));
+        const sk: BigSecretKey = { n, d, p, q };
+
+        // 1. m = bytes_to_int(blinded_msg)
+        const m = os2ip(blindMsg);
+
+        // 2. sk_derived, pk_derived = DeriveKeyPair(sk, info)
+        const { secretKey: sk_derived, publicKey: pk_derived } = await this.deriveKeyPair(sk, info);
+
+        if (this.params.supportsRSARAW) {
+            const { privateKey } = await PartiallyBlindRSA.bigKeyPairToCryptoKeyPair(
+                { secretKey: sk_derived, publicKey: pk_derived },
+                {
+                    modulusLength: kLen * 8,
+                    publicExponent: new Uint8Array([1, 0, 1]),
+                    hash: this.params.hash,
+                },
+                true,
+            );
+            return rsaRawBlingSign(privateKey, blindMsg);
+        }
+
+        // 3. s = RSASP1(sk_derived, m)
+        const s = rsasp1(sk_derived, m);
+
+        // 4. m' = RSAVP1(pk_derived, s)
+        const mp = rsavp1(pk_derived, s);
+
+        // 5. If m != m', raise "signing failure" and stop
+        if (m.equals(mp) === false) {
+            throw new Error('signing failure');
+        }
+
+        // 6. blind_sig = int_to_bytes(s, kLen)
+        // 7. output blind_sig
+        return i2osp(s, kLen);
```

Comment:
here the function is returning the signature, and skipped the verification.
Note that in the specification, the signature is always checked, just in case the generation had produced an error.

Reply:
I'm not sure I totally get your point. The spec does not seem to mandate checking https://datatracker.ietf.org/doc/html/draft-amjad-cfrg-partially-blind-rsa-02#name-blindsign

The only thing that differs is that if the bitlength of the signature is bigger than the modulus length, int_to_bytes will error with a message different from the one defined by the spec.

Reply:
the question here is whether `rsaRawBlingSign` processes these steps:

```
4. m' = RSAVP1(pk_derived, s)
5. If m != m', raise "signing failure" and stop
```

Reply:
reading the [platform code](https://github.com/cloudflare/workerd/blob/main/src/workerd/api/crypto-impl-asymmetric.c%2B%2B#L838-L879), I'm tempted to say no

we'll need to update [blind rsa](https://www.rfc-editor.org/rfc/rfc9474#name-blindsign) implementation as well, given the same pattern is used.